frontend/packages/console-app/src/components/cloud-shell/CloudShellTerminal.tsx

```diff
@@ -28,11 +25,17 @@ type Props = {
 type CloudShellTerminalProps = StateProps & Props;
 
 const CloudShellTerminal: React.FC<CloudShellTerminalProps> = ({ user, onCancel }) => {
-  const [namespace, setNamespace] = React.useState(getCloudShellNamespace());
+  const [namespace, setNamespace, isNamespaceLoaded] = useUserSettingsCompatibility(
+    CLOUD_SHELL_NAMESPACE_CONFIG_STORAGE_KEY,
+    CLOUD_SHELL_NAMESPACE,
+  );
```

Comment:
The way the default namespace works, we must have it loaded before we initialize the `useCloudShellWorkspace` hook. Which means this must go in a separate component like an HOC.

We can either create an HOC now for others to use or make a one off case and create a parent component which doesn't render the child component until loaded.